src/linear_algebra/determinant.lean

```diff
@@ -75,6 +76,16 @@ begin
   simp [det, card_eq_zero.mp h, perm_eq],
 end
 
+lemma det_eq_elem_of_card_eq_one {A : matrix n n R} (h : fintype.card n = 1) (k : n) :
+  det A = A k k :=
+begin
+  have h1 : (univ : finset (perm n)) = {1},
+  { apply univ_eq_singleton_of_card_one (1 : perm n),
+    simp [card_univ, fintype.card_perm, h] },
+  have h2 := univ_eq_singleton_of_card_one k h,
+  simp [det, h1, h2],
+end
+
```

Comment:
The following should work as a `simp` lemma. It looks like there isn't much connecting `fintype.card n = 1` and `unique n`, so it would probably add both and just leave `det_unique` unused for now.
```lean
/-- If `n` has only one element, the determinant of an `n` by `n` matrix is just that element.

Although `unique` implies `decidable_eq` and `fintype`, the instances might
not be syntactically equal. Thus, we need to fill in the args explicitly. -/
@[simp]
lemma det_unique {n : Type*} [unique n] [d : decidable_eq n] [f : fintype n] (A : matrix n n R) :
  @det _ d f _ _ A = A (default n) (default n) :=
by simp [det, univ_unique]
```

Reply:
Are you sure you have to actually fill in the args explicitly? I'd have thought simply the presence of the instance in the argument list is enough.

Reply:
You are right, instance parameters take precedence over global instances. I also forgot to include the instances needed to make the proof by `simp` work, so the full addition should be:

```lean
instance {n : Type*} [unique n] : unique (equiv.perm n) :=
{ default := 1,
uniq := λ σ, equiv.ext (λ i, subsingleton.elim _ _) }

@[simp] lemma default_perm {n : Type*} : default (equiv.perm n) = 1 := rfl

/-- If `n` has only one element, the determinant of an `n` by `n` matrix is just that element.

Although `unique` implies `decidable_eq` and `fintype`, the instances might
not be syntactically equal. Thus, we need to fill in the args explicitly. -/
@[simp]
lemma det_unique {n : Type*} [unique n] [d : decidable_eq n] [f : fintype n] (A : matrix n n R) :
  det A = A (default n) (default n) :=
by simp [det, univ_unique]
```

Reply:
So I guess you want to promote https://leanprover-community.github.io/mathlib_docs/data/equiv/basic.html#equiv.perm_subsingleton to `unique`?

Reply:
Add a new instance rather than promoting the other one, but yes. (`fin 0` is a `subsingleton` but not `unique` since there is no `default : fin 0`)